pkg/invariants/incluster_disruption_serializer/remote.go

```diff
@@ -381,18 +255,177 @@ func fetchEventsFromFileOnNode(ctx context.Context, clientset *kubernetes.Client
-		if !strings.HasPrefix(event.Locator, "disruption/") {
+	fmt.Fprintf(os.Stderr, "Fetched %d events from node %s\n", len(events), nodeName)
+	return events, err
+}
+
+type InvariantInClusterDisruption struct {
+	adminRESTConfig *rest.Config
+}
+
+func NewInvariantInClusterDisruption() invariants.InvariantTest {
+	return &InvariantInClusterDisruption{}
+}
+
+func (i *InvariantInClusterDisruption) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, _ monitorapi.RecorderWriter) error {
+	fmt.Fprintf(os.Stderr, "Starting in-cluster monitoring daemonsets\n")
+	i.adminRESTConfig = adminRESTConfig
+	kubeClient, err := kubernetes.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	configClient, err := configclient.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	internalAPI, err := url.Parse(infra.Status.APIServerInternalURL)
+	if err != nil {
+		return err
+	}
+	apiIntHost := internalAPI.Hostname()
+
+	err = createNamespace(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createServiceAccount(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createRBACPrivileged(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createMonitorRole(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createMonitorCRB(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createServiceNetworkDS(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createLocalhostDS(ctx, kubeClient)
+	if err != nil {
+		return err
+	}
+	err = createInternalLBDS(ctx, kubeClient, apiIntHost)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *InvariantInClusterDisruption) CollectData(ctx context.Context, storageDir string, beginning time.Time, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	fmt.Fprintf(os.Stderr, "Stopping in-cluster monitoring daemonsets\n")
+	client, err := kubernetes.NewForConfig(i.adminRESTConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Start watcher before daemonset removal request to make sure delete event is not skipped
+	c := make(chan error)
+	timeLimitedCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+	go func() {
+		_, watchErr := watchtools.UntilWithSync(timeLimitedCtx,
+			cache.NewListWatchFromClient(
+				client.AppsV1().RESTClient(), "daemonsets", namespace, fields.Everything()),
+			&appsv1.DaemonSet{},
+			nil,
+			func(event watch.Event) (bool, error) {
+				return event.Type == watch.Deleted, nil
+			},
+		)
+		if watchErr != nil {
+			watchErr = fmt.Errorf("daemonsets in namespace %s didn't get destroyed: %v", namespace, watchErr)
+		}
+		c <- watchErr
+	}()
+
+	// Remove daemonsets first to avoid trailing false-positive disruption intervals
+	dsClient := client.AppsV1().DaemonSets(namespace)
```

Comment:
If you did that, then this delete is actually to be done in cleanup as a simple ns delete.

Reply:
Yes, Cleanup is now much simpler - but it also needs to remove ClusterRoleBinding and ClusterRole

Reply:
Just delete the namespace and then remove whatever cluster scope things you need.